enterprise-suite/es-monitor-api/prometheus.yml

```diff
@@ -171,48 +175,109 @@ scrape_configs:
+  # * `{{ .PrometheusDomain }}/port`: If the metrics are exposed on a different port to the
+  # service then set this appropriately.
 
+  - job_name: 'kubernetes-services'
     kubernetes_sd_configs:
-      - role: service
+      - role: endpoints
+
+    honor_labels: true
 
     relabel_configs:
-      - source_labels: [__meta_kubernetes_service_annotation_{{ dot2underscore .PrometheusDomain }}_probe]
+      # these have dedicated scrape sections:
+      - source_labels: [__meta_kubernetes_endpoints_name]
+        regex: prometheus-node-exporter
+        action: drop
+      - source_labels: [__meta_kubernetes_endpoints_name]
+        regex:  prometheus-kube-state-metrics
+        action: drop
+      # _scrape _scheme _path _port annotation triggers:
+      - source_labels: [__meta_kubernetes_service_annotation_{{ dot2underscore .PrometheusDomain }}_scrape]
         action: keep
         regex: true
-      - source_labels: [__address__]
-        target_label: __param_target
-      - target_label: __address__
-        replacement: blackbox
-      - source_labels: [__param_target]
-        target_label: instance
+      - source_labels: [__meta_kubernetes_service_annotation_{{ dot2underscore .PrometheusDomain }}_scheme]
+        action: replace
+        target_label: __scheme__
+        regex: (https?)
+      - source_labels: [__meta_kubernetes_service_annotation_{{ dot2underscore .PrometheusDomain }}_path]
+        action: replace
+        target_label: __metrics_path__
+        regex: (.+)
+      - source_labels: [__address__, __meta_kubernetes_service_annotation_{{ dot2underscore .PrometheusDomain }}_port]
+        action: replace
+        target_label: __address__
+        regex: (.+)(?::\d+);(\d+)
+        replacement: $1:$2
+      # metadata labels
       - action: labelmap
         regex: __meta_kubernetes_service_label_(.+)
       - source_labels: [__meta_kubernetes_namespace]
         target_label: namespace
+      - source_labels: [__meta_kubernetes_namespace]
+        target_label: kubernetes_namespace
       - source_labels: [__meta_kubernetes_service_name]
+        action: replace
         target_label: kubernetes_name
+      - source_labels: [__meta_kubernetes_pod_node_name]
+        action: replace
+        target_label: node_name
+      - source_labels: [__meta_kubernetes_pod_host_ip]
+        action: replace
+        target_label: node_ip
+      - action: labelmap
+        regex: __meta_kubernetes_pod_label_(.+)
+      - source_labels: [__meta_kubernetes_pod_name]
+        action: replace
+        target_label: kubernetes_pod_name
+      - source_labels: [__meta_kubernetes_pod_label_pod_template_hash, __meta_kubernetes_pod_name, workload]
+        action: replace
+        regex: '[^;]+;(.*)-[^-]+-[^-]+;'
+        target_label: workload
+      - source_labels: [__meta_kubernetes_pod_label_statefulset_kubernetes_io_pod_name, workload]
+        action: replace
+        regex: '(.*)-[0-9]+;'
+        target_label: workload
```

Comment:
There is a lot of duplication with the pod SD section. Would it make sense to stick this in a common file and include it in both places?

Reply:
prometheus config doesn't have that feature, this is just a file and is not evaluated as a helm chart, so.... don't think there's a easy shot for that one